src/server/main_service.cc

```diff
@@ -2201,6 +2215,13 @@ void Service::Pubsub(CmdArgList args, ConnectionContext* cntx) {
     PubsubChannels(pattern, cntx);
   } else if (subcmd == "NUMPAT") {
     PubsubPatterns(cntx);
+  } else if (subcmd == "NUMSUB") {
+    if (args.size() == 1) {
+      (*cntx)->SendEmptyArray();
+      return;
+    }
+
+    PubsubNumSub(args, cntx);
```

Comment:
I'd do `args.remove_prefix(1)` to remove the sub-command argument, which will make for a better encapsulation and self-containment of `PubsubNumSub()`

Reply:
fixed: https://github.com/dragonflydb/dragonfly/pull/2282/commits/87370ddd02514b5eecd8c91f8aefcd86852ae19e